### PR TITLE
Modify metric endpoints

### LINKED ISSUE #1620: Proposal: modify metrics endpoints to make more similar
To define data sources we have the following:

- **HTTP**:
  - URL: `http://iter8.default:8080/httpDashboard`
  - query string: `namespace=ns&experiment=test_name`
- **gRPC**:
  - URL: `http://iter8.default:8080/grpcDashboard`
  - query string: `namespace=ns&experiment=test_name`
- **A/B/n**:
  - URL: `http://iter8.default:8080/metrics`
  - query string: `application=ns%2Fapp`

I wondered if we wanted to regularize this:
- For A/B/n use `abnDashboard` instead of `metrics` as the endpoint
- For A/B/n change structure of query string to be `namespace=ns&application=app`
- change `experiement` to `test` or `performance_test`
- Not sure how to make the second parameter the same; but there is no real need to

### PR DESCRIPTION
**Pull request by Alan-Cha**
Fix #1620 

Replace #1628

### CODE REVIEW

File: metrics/server_test.go [hunk 1]
```diff
@@ -655,8 +655,8 @@ func TestGetMetrics(t *testing.T) {
 	w := httptest.NewRecorder()
 	rm := allRoutemaps.GetAllRoutemaps().GetRoutemapFromNamespaceName("default", "test")
 	assert.NotNil(t, rm)
-	req := httptest.NewRequest(http.MethodGet, "/metrics?application=default%2Ftest", nil)
-	getMetrics(w, req)
+	req := httptest.NewRequest(http.MethodGet, util.AbnDashboard+"?application=default%2Ftest", nil)
```
Comment (Alan-Cha) [inline metrics/server_test.go]: Hmmm actually this query string was not modified and the test did not fail. Will look into this.

File: base/metrics.go [hunk 2]
```diff
@@ -14,11 +14,11 @@ const (
 	// MetricsServerURL is the URL of the metrics server
 	MetricsServerURL = "METRICS_SERVER_URL"
 
-	// MetricsPath is the path to the GET /metrics endpoint
-	MetricsPath = "/metrics"
-
 	// ExperimentResultPath is the path to the PUT /experimentResult endpoint
 	ExperimentResultPath = "/experimentResult"
```
Comment (kalantar) [inline base/metrics.go]: should this also be changed to `testResult` ?

File: metrics/server.go [hunk 3]
```diff
@@ -193,17 +192,27 @@ func getMetrics(w http.ResponseWriter, r *http.Request) {
...
 	}
-	log.Logger.Tracef("getMetrics called for application %s", application)
+
+	namespaceApplication := fmt.Sprintf("%s/%s", namespace, application)
+
+	log.Logger.Tracef("getAbnDashboard called for application %s", namespaceApplication)
 
 	// identify the routemap for the application
-	namespace, name := util.SplitApplication(application)
+	namespace, name := util.SplitApplication(namespaceApplication)
```
Comment (kalantar) [inline metrics/server.go]: this shouldn't be necessary any more. namespace name are namespace, application from a few lines above.

**Review by kalantar — DISMISSED** [review]

File: metrics/server.go (hunk 3 above)
Comment (Alan-Cha) [reply metrics/server.go]: Oop sorry

File: metrics/server.go [hunk 4]
```diff
@@ -724,10 +732,10 @@ func getGRPCDashboard(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write(dashboardBytes)
 }
 
-// putExperimentResult handles PUT /experimentResult with query parameter application=namespace/name
+// putExperimentResult handles PUT /testResult with query parameter test=name and namespace=namespace
```
Comment (Alan-Cha) [inline metrics/server.go:735]: Note gap in function name `putExperimentResult` and endpoint `/testResult with query parameter test=name and namespace=namespace`

File: metrics/server.go [hunk 5]
```diff
@@ -742,13 +750,13 @@ func putExperimentResult(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	experiment := r.URL.Query().Get("experiment")
+	experiment := r.URL.Query().Get("test")
```
Comment (Alan-Cha) [inline metrics/server.go:753]: Note gap here

**Review by kalantar — COMMENTED** [review]
lgtm

**Review by kalantar — APPROVED** [review]

**Pull request merged**